### DISCUSSION TITLE
Tune function - run on entire train set prior to cross-validation?

### DISCUSSION THREAD

**Discussion by shlid007** [kogalur/randomForestSRC#399] [Q&A]
I know nested cross-validation can be done as part of hyperparameter tuning, but is another acceptable approach to use the "tune" function on the entire training set PRIOR to cross-validation?

**Comment by ishwaran** [comment]
Using the function `tune` is not equivalent to nested cross-validation.  The latter is a gold-standard method to avoid contaminating  model performance evaluation with tuning of hyper-parameter selection.   That being said, nested cross-validation is not feasible in many problems (especially using random forests) so we can consider `tune` as a fast approximate way to accomplish hyper-parameter evaluation with minimal contamination.
